src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -5534,7 +5541,7 @@ public String getJsonLd() {
                 return jsonLd;
             } else {
                 logger.fine("No cached schema.org JSON-LD available. Going to the database.");
-                String jsonLdProduced = workingVersion.getJsonLd();
+                String jsonLdProduced = instance.getExportAsString(workingVersion.getDataset(), SchemaDotOrgExporter.NAME);
```

Comment:
This is going to get the schema.org for the latest version rather than the one being viewed. I think what's needed is a method in ExportService that allows you to specify the version.
Also - the datasetVersion.getJsonLd() method used to cache the output in memory for reuse. The Exporter currently only writes the schema.org for the latest version to disk (which we may still want) and if we don't cache there, we may still want to store the jsonLd in memory. Perhaps keeping the DatasetVersion.getJsonLd() method (which would cache) but calling a new version specific method in the exporter would handle that?

Reply:
FWIW: https://github.com/IQSS/dataverse/blob/803a6e94176a8114a96aa06e8c7369aab658731f/src/main/java/edu/harvard/iq/dataverse/DatasetPage.java#L5529 means the schema.org is only in the page header for the latest version so nominally we don't currently need to keep this method version specific for that reason. That said, other work will need it so removing it now only to have to bring it back would just add work.